src/mercury_engine_data_structures/exporters/raw_texture.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import annotations
+
+import dataclasses
+import math
+from typing import TYPE_CHECKING
+
+import py_tegra_swizzle
+
+from mercury_engine_data_structures.formats.bctex import Bctex, BlockType, XTX_Tegra_Format
+from mercury_engine_data_structures.game_check import Game
+
+if TYPE_CHECKING:
+    from construct.core import Container
+
+
+def div_round_up(n: int, d: int) -> int:
+    return (n + d - 1) // d
+
+
+@dataclasses.dataclass
+class Surface:
+    width: int
+    height: int
+    data: bytes
+
+
+@dataclasses.dataclass
+class Texture2D:
+    mips: list[Surface]
+
+    def mip0(self) -> Surface:
+        return self.mips[0]
+
+
+@dataclasses.dataclass
+class Array:
+    width: int
+    height: int
+    format: XTX_Tegra_Format
+    members: list[Texture2D]
+
+
+class RawTexture:
+    bctex: Bctex
+    name: str
+    textures: list[Array]
+
+    def __init__(self, texture: Bctex) -> None:
+        if texture.target_game != Game.DREAD:
+            raise ValueError("Only Dread bctex can be exported!")
+
+        self.bctex = texture
+        self.name = texture.raw.data.name
+
+        self.parse()
+
+    def parse(self):
```

Comment:
Is this method supposed to be called multiple times? If it's just the constructor, make it private.